src/map.cpp

```diff
@@ -5652,6 +5656,79 @@ void map::draw( const catacurses::window &w, const tripoint &center )
     }
 }
 
+void map::update_terrain_weather()
+{
+    const w_point weatherPoint = *g->weather_precise;
+    if( g->weather == WEATHER_FLURRIES ) {
+        snowfall += 2;
+    } else if( g->weather == WEATHER_SNOW ) {
+        snowfall += 4;
+    } else if( g->weather == WEATHER_SNOWSTORM ) {
+        snowfall += 8;
+    } else {
+        snowfall -= std::max( 0, ( static_cast<int>( weatherPoint.temperature - 32 ) ) );
+    }
```

Comment:
Can be converted to a `switch` statement inside a constant function that returns increments.
```c++
switch( g->weather ) {
case WEATHER_FLURRIES:
    return 2;
case WEATHER_SNOW:
    ...
}
```